src/compile/data/optimizers.ts

```diff
@@ -11,14 +11,12 @@ import {TimeUnitNode} from './timeunit';
 import {CalculateNode, FilterNode} from './transforms';
 
 /**
- * Start optimization path at the leaves. Useful for merging up things.
+ * Start optimization path at the leaves. Useful for merging up or removing things.
  */
-export function optimizeFromLeaves(f: (node: DataFlowNode) => void) {
+export function iterateFromLeaves(f: (node: DataFlowNode) => void) {
```

Comment:
@domoritz  I wonder why you have to change the logic here since this function seems unused. 

This cause some conflicts with https://github.com/vega/vega-lite/pull/2168/commits/29ebae677239d2048dab7a99c8010ad516704570 in #2168. 
(I also wonder if I still need that `removeUnusedSubtrees` logic?)